### PR TITLE
Add links to pages, translations and automatic redirect when accessin…

### PR DESCRIPTION
**Pull request by liviuignat**
- Important: This PR does not include the changes related to the "dynamic links" that we discussed in the previous PR. This branch was created before the previous PR was reviewed, based on that branch. A new PR will follow.
- Important 2: Changed application port to __9200__
- Add `loggedInRedirectUrl` with redirect logic on the index page.
- Moved some route logic from `/server/controllers/index.ts` to `/server/controllers/siteController.ts`
- Updated `tslint.json` validation rules for unused variables and removed unused variables.
- Update `FormattedLink` component to accept `id` (for translation id as link) or `href` (for direct link but with locale appended in front of the link).

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
-  port: process.env.PORT || 3000,
+  port: process.env.PORT || 9200,
+  loggedInRedirectUrl: 'http://dev.everreal.co:9300/properties/decision',
```
Comment (liviuignat) [inline config/default.js]: This link will decide where to redirect the user next, depending on his registration state. Eg: after registration, did he finished setting up his account or not - like giving us all the information we initially require when he is owner, property manager, tenant, etc.

File: src/universal/common/components/FormattedLink/FormattedLink.tsx [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
-  const {intl: {formatMessage}} = context;
-  const {id, children} = props;
-  const formattedHref = formatMessage({id});
+  const {intl: {formatMessage, locale}} = context;
+  const {id, href, children} = props;
+  let formattedHref = '';
+  if (id) {
+    formattedHref = formatMessage({id});
+  } else if (href) {
+    const urlParts = [locale, ...href.split('/')].filter(part => !!part);
```
Comment (liviuignat) [inline src/universal/common/components/FormattedLink/FormattedLink.tsx]: If `href` is present, will form a new url appended with the locale and non-empty parts of the url.

**Comment by liviuignat** [comment]
Hi @RaulTsc , this PR does not include the changes we discussed about the links [here](https://github.com/liviuignat/heimdall/pull/19#discussion-diff-97396968R21). The branch was created before our discussion, but waited for a review.

I will create another branch for that. I have some ideas, will create a PR and discuss more :)

**Comment by liviuignat** [comment]
@RaulTsc, I will have merge this. If you have any comments, I will create a new PR to solve them :).

File: config/default.js (hunk 1 above)
Comment (RaulTsc) [inline config/default.js]: 👍

File: src/universal/auth/pages/RegisterPage/RegisterPage.tsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ export default class RegisterPage extends Component<any, any> {
 
   public async handleSubmit(data: IUser) {
     if (data && data.email) {
```
Comment (RaulTsc) [inline src/universal/auth/pages/RegisterPage/RegisterPage.tsx]: Does this need to be async? Why is it different than the handleSubmit in LoginPage.tsx?

File: src/universal/common/components/FormattedLink/FormattedLink.tsx (hunk 2 above)
Comment (RaulTsc) [inline src/universal/common/components/FormattedLink/FormattedLink.tsx]: 👍

File: tslint.json [hunk 4]
```diff
@@ -7,12 +7,13 @@
    */
   "extends": ["tslint:recommended", "tslint-react"],
   "rules": {
+    "no-unused-variable": [true, "react"],
```
Comment (RaulTsc) [inline tslint.json]: 👍  I also saw some unused variables before.

File: src/universal/auth/pages/RegisterPage/RegisterPage.tsx (hunk 3 above)
Comment (liviuignat) [inline src/universal/auth/pages/RegisterPage/RegisterPage.tsx]: it's calling `await registerUserAction`, but it's not required, I was waiting for the response in the original version.

**Pull request merged**